### PR TITLE
[Chat] History API

### PR DESCRIPTION
**Pull request by flypaper0**
# Description
- HistoryClient
- HistoryClient integration tests
- History integrated into chat sdk
- KeyedDatabase refactor, legacy store removed
- Messages batch update

Resolves # (issue)

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Example/IntegrationTests/History/HistoryTests.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        try await historyClient.registerTags(
+            payload: RegisterPayload(tags: [String(tag)], relayUrl: relayUrl),
+            historyUrl: historyUrl)
+
+        try await relayClient2.subscribe(topic: topic)
+        try await relayClient1.publish(topic: topic, payload: payload, tag: tag, prompt: false, ttl: 3000)
+
+        wait(for: [exp], timeout: InputConfig.defaultTimeout)
+
+        sleep(5) // History server has a queue
```
Comment (llbartekll) [inline Example/IntegrationTests/History/HistoryTests.swift:67]: may the test fail sometimes due to insufficient delay?

File: Example/IntegrationTests/History/HistoryTests.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import Foundation
+import Combine
+import XCTest
+@testable import WalletConnectHistory
+
+final class HistoryTests: XCTestCase {
```
Comment (llbartekll) [inline Example/IntegrationTests/History/HistoryTests.swift:6]: please exclude those tests from smoke test plan

File: Sources/Chat/ProtocolServices/History/HistoryService.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    func fetchMessageHistory(thread: Thread) async throws -> [Message] {
+        let wrappers: [MessagePayload.Wrapper] = try await historyClient.getMessages(
+            topic: thread.topic,
+            count: 200, direction: .backward
+        )
+
+        return wrappers.map { wrapper in
+            let (messagePayload, messageClaims) = try! MessagePayload.decodeAndVerify(from: wrapper)
```
Comment (llbartekll) [inline Sources/Chat/ProtocolServices/History/HistoryService.swift:24]: should we force unwrap here?
would make sense to compactMap nil values or throw an error instead?

**Review by llbartekll — APPROVED** [review]

File: Example/IntegrationTests/History/HistoryTests.swift (hunk 1 above)
Comment (flypaper0) [reply Example/IntegrationTests/History/HistoryTests.swift:67]: removed from smoke tests

**Pull request merged**